### PR TITLE
Add protobufs for credential and vault packages

### PR DESCRIPTION
**Pull request by mgaffney**

### CODE REVIEW

**Review by louisruch — COMMENTED** [review]
A typo in the comments, but your `filed numbers` in the proto files has raised a question.  I will discuss with the team in slack

File: internal/proto/local/controller/storage/credential/store/v1/credential.proto [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+// Package store provides protobufs for storing types in the credential package.
+package controller.storage.credential.store.v1;
+option go_package = "github.com/hashicorp/boundary/internal/credential/store;store";
+
+import "controller/storage/timestamp/v1/timestamp.proto";
+import "controller/custom_options/v1/options.proto";
+
+message Store {
+  // public_is is a surrogate key suitable for use in a public API.
```
[the remark this answers is not in the document]
Comment (mgaffney) [reply internal/proto/local/controller/storage/credential/store/v1/credential.proto]: ty!

File: internal/proto/local/controller/storage/credential/store/v1/credential.proto [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  // @inject_tag: `gorm:"primary_key"`
+  string public_id = 1;
+
+  // The scope_id of the owning scope and must be set.
+  // @inject_tag: `gorm:"not_null"`
+  string scope_id = 2;
+}
+
+message Library {
+  // public_is is a surrogate key suitable for use in a public API.
```
[the remark this answers is not in the document]
Comment (mgaffney) [reply internal/proto/local/controller/storage/credential/store/v1/credential.proto]: ty!

**Review by louisruch — APPROVED** [review]
Given the discussion LGTM

**Pull request merged**